File: pset6/cash/cash.py
# Program that calculates how many coins are needed to return the requested amount of change
from cs50 import get_float


def main():
    # Get input from the user
    while True:
        n = get_float("Change owned: ")
        if n >= 0:
            break
    # Round the float input to an integer
    cents = round(n * 100)

    # Initialize quarters, dimes, nickels, pennies and coins
    q = 0
    d = 0
    n = 0
    p = 0
    coins = 0

    # Check for quarters
    while (cents >= 25):
        q += 1
        coins += 1
        cents = cents - 25
    print(f"Quarters: {q}")
    print(f"Cents after quarters: {cents}")

    # Check for dimes
    while (cents >= 10):
        d += 1
        coins += 1
        cents = cents - 10
    print(f"Dimes: {d}")
    print(f"Cents after dimes: {cents}")

    # Check for nickels
    while (cents >= 5):
        n += 1
        coins += 1
        cents = cents - 5
    print(f"Nickels: {n}")
    print(f"Cents after nickels: {cents}")

    # Check for pennies
    while (cents >= 1):
        p += 1
        coins += 1
        cents = cents - 1
    print(f"Pennies: {p}")
    print(f"Cents after pennies: {cents}")

    print(f"Total coins: {coins}")


main()

File: pset1/cash/cash.c
// Program that runs a basic greedy algorith. 
// User inputs the change owed in Dollars, program outputs the ammount of coins used.
// Quarters: 0.25, Dimes: 0.10, Nickels: 0.05, Pennies: 0.01

// Conditions and declarations
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)

{
    // Prompt user for input
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);

    // Convert float to int, making input in cents
    int cents = round(dollars * 100);
    int coins = 0;

    // Loop for quarters
    for (int q = 1; (cents / 25 >= 1); q++)
    {
        cents = cents - 25;
        coins++;
    }

    // Loop for dimes
    for (int d = 1; cents / 10 >= 1; d++)
    {
        cents = cents - 10;
        coins++;
    }

    // Loop for nickels
    for (int n = 1; cents / 5 >= 1; n++)
    {
        cents = cents - 5;
        coins++;
    }

    // Loop for pennies
    for (int p = 1; cents / 1 >= 1; p++)
    {
        cents = cents - 1;
        coins++;
    }

    // Prints number of coins used
    printf("Total coins used: %i\n", coins);
}